Upgrade wildfly9 & Adding support for implementing external database providers.

fharms opened the pull request:
- This is the first step to support other providers like
  flywaydb by separate the database logic from the 
  dbBootstrapScanDetectorProcessor into it's own class
- Refactor annotation scanner to use Wildfly annotation
  scanner feature instead for better performance. This also 
  remove the dependencies to the scannotation 3. party lib
  and makes the code cleaner

fharms committed:
add idea project files

fharms committed:
fixed untracked files

fharms committed:
Upgrade to Wildfly 9

Upgrade build files to match dependencies for Wildfly 9
and refactored deprecated APIs to follow new API's
guidelines

fharms committed:
Adding support for implementing external database providers.

- This is the first step to support other providers like
flywaydb by separate the database logic from the
dbBootstrapScanDetectorProcessor into it's own class

- Refactor annotation scanner to use Wildfly annotation
scanner feature instead for better performance. This also
remove the dependencies to the scannotation 3. party lib
and makes the code cleaner

fharms committed:
Fixed multiple rebase errors

fharms committed:
Bump the version number

fharms commented on integrate/src/main/resources/docs/schema/wildfly-bootstrap-2.0.xsd:
```diff
@@ -0,0 +1,72 @@
+        </xsd:annotation>
+        <xsd:sequence>
+            <xsd:element name="hibernate" type="hibernateProviderType" minOccurs="0" maxOccurs="1" />
+            <xsd:element name="flywaydb" type="flywaydbProviderType" minOccurs="0" maxOccurs="1" />
```
Remove this for now, it's not part of this release

fharms commented on extension/src/main/java/org/wildfly/extras/db_bootstrap/DbBootstrapScanDetectorProcessor.java:
```diff
@@ -370,15 +261,11 @@ private AnnotationDB scanForAnnotation(Set<URL> jars) throws IOException, URISyn
+
         for (VirtualFile virtualFile : entries) {
             try {
-                URL url = VFSUtils.getRootURL(virtualFile);
```
Investigate if this was a mistake it was removed?

fharms commented on extension/src/main/java/org/wildfly/extras/db_bootstrap/DbBootstrapScanDetectorProcessor.java:
```diff
@@ -103,109 +96,90 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
-                throw new DeploymentUnitProcessingException(e);
-            }
-            long duration = System.currentTimeMillis() - before;
-            DbBootstrapLogger.ROOT_LOGGER.infof("Database bootstrapping took [%s] ms", duration);
```
Need to still time how long it take

fharms committed:
- Remove not used scheme definition
- Add log of execution time

ctomc commented on extension/src/main/java/org/wildfly/extras/db_bootstrap/DbBootstrapSubsystemParser2_0.java:
```diff
@@ -0,0 +1,64 @@
+/**
+ * @author Flemming Harms
+ */
+class DbBootstrapSubsystemParser2_0 implements XMLStreamConstants, XMLElementReader<List<ModelNode>>, XMLElementWriter<SubsystemMarshallingContext> {
```
you could extend PersistentResourceXMLParser it is much cleaner to implement :)

fharms committed:
- Refactor annotation scanner to use Wildfly annotation
scanner feature instead for better performance.

- Refactor the logic to create Session with the new API

fharms commented on README.md:
```diff
@@ -13,8 +13,9 @@ This extension is currently aimed at Hibernate users. But it can also be used wi
 
 Prerequisite
```
Remember to update the installation procedure after the module configuration is moved

fharms committed:
- Update README.md
